File: frontend/src/types/TerminDTOWithFilmDTOSlideshow.ts
import {FilmDTOForm} from "./FilmDTOForm.ts";

// this one mirrors TerminDTOWithFilmDTOSlideshow
interface TerminDTOWithFilmDTOSlideshow {
    tnr: number;
    vorstellungsbeginn?: string;
    titel?: string | null;
    kurztext?: string | null;
    besonderheit?: string| null;
    bild?: string| null;
    sonderfarbe?: number | null;
    veroeffentlichen?: number | null;
    mainfilms: FilmDTOForm[];
}

export default TerminDTOWithFilmDTOSlideshow;

File: backend/src/main/java/org/pupille/backend/mysql/screening/TerminDTOWithFilmDTOSlideshow.java
package org.pupille.backend.mysql.screening;

import org.pupille.backend.mysql.film.Film;
import org.pupille.backend.mysql.film.FilmDTOForm;
import org.pupille.backend.mysql.termin.Termin;

import java.time.LocalDateTime;
import java.util.List;

public record TerminDTOWithFilmDTOSlideshow(
        Long tnr,
        LocalDateTime vorstellungsbeginn,
        String titel,
        String kurztext,
        String besonderheit,
        String bild,
        Integer sonderfarbe,
        Short veroeffentlichen,
        List<FilmDTOForm> mainfilms
) {
    public TerminDTOWithFilmDTOSlideshow(Termin termin, List<Film> films) {
        this(
                termin.getTnr(),
                termin.getVorstellungsbeginn(),
                termin.getTitel(),
                termin.getKurztext(),
                termin.getBesonderheit(),
                termin.getBild(),
                termin.getSonderfarbe(),
                termin.getVeroeffentlichen(),
                films.stream()
                        .map(FilmDTOForm::new)
                        .toList()
        );
    }
}